topology/probes/k8s/cluster.go

```diff
@@ -27,47 +27,79 @@ import (
 	"github.com/skydive-project/skydive/probe"
 	"github.com/skydive-project/skydive/topology"
 	"github.com/skydive-project/skydive/topology/graph"
-
-	"k8s.io/client-go/kubernetes"
 )
 
 // ClusterName is the name of the k8s cluster
 const ClusterName = "cluster"
 
 var clusterEventHandler = graph.NewEventHandler(100)
 
+var clusterNode *graph.Node
+
 type clusterCache struct {
 	*graph.EventHandler
 	graph *graph.Graph
 }
 
-func (c *clusterCache) Start() {
+func (c *clusterCache) addClusterNode() {
 	c.graph.Lock()
 	defer c.graph.Unlock()
 
 	m := NewMetadata(Manager, "cluster", nil, ClusterName)
-	node := c.graph.NewNode(graph.GenID(), m, "")
-	c.NotifyEvent(graph.NodeAdded, node)
+	clusterNode = c.graph.NewNode(graph.GenID(), m, "")
+	c.NotifyEvent(graph.NodeAdded, clusterNode)
 	logging.GetLogger().Debugf("Added cluster{Name: %s}", ClusterName)
 }
 
+func (c *clusterCache) Start() {
+}
+
 func (c *clusterCache) Stop() {
 }
 
-func newClusterProbe(clientset *kubernetes.Clientset, g *graph.Graph) Subprobe {
-	return &clusterCache{
+func newClusterProbe(clientset interface{}, g *graph.Graph) Subprobe {
+	c := &clusterCache{
 		EventHandler: clusterEventHandler,
 		graph:        g,
 	}
+	c.addClusterNode()
+	return c
+}
+
+type clusterLinker struct {
+	*graph.ResourceLinker
+	g             *graph.Graph
+	objectIndexer *graph.MetadataIndexer
+}
+
+func (linker *clusterLinker) createEdge(cluster, object *graph.Node) *graph.Edge {
+	id := graph.GenID(string(cluster.ID), string(object.ID))
+	return linker.g.CreateEdge(id, cluster, object, topology.OwnershipMetadata(), graph.TimeUTC(), "")
+}
+
+// GetABLinks returns all the incoming links for a node
+func (linker *clusterLinker) GetABLinks(objectNode *graph.Node) (edges []*graph.Edge) {
+	return
+}
+
+// GetBALinks returns all the incoming links for a node
+func (linker *clusterLinker) GetBALinks(objectNode *graph.Node) (edges []*graph.Edge) {
+	edges = append(edges, linker.createEdge(clusterNode, objectNode))
+	return
 }
 
 func newClusterLinker(g *graph.Graph, manager string, types ...string) probe.Probe {
-	clusterIndexer := graph.NewMetadataIndexer(g, clusterEventHandler, graph.Metadata{"Manager": Manager, "Type": "cluster"})
+	clusterIndexer := newObjectIndexerFromFilter(g, g, newTypesFilter(Manager, "cluster"))
```

Comment:
This doesn't seem to be used any more. This can be removed an nil be passed to NewResourceLinker instead

Reply:
ok